resip/stack/ssl/Security.hxx

```diff
@@ -181,7 +181,7 @@ class BaseSecurity
       // retrieves a list of all certificate names (subjectAltNAme's and CommonName)
       static void getCertNames(X509 *cert, std::list<PeerName> &peerNames, bool useEmailAsSIP = false);
 
-      static bool isSelfSigned(const X509* cert);
+      static bool isSelfSigned(X509* cert);
```

Comment:
This should not be needed. The operation is not supposed to modify the certificate and both `X509_get_issuer_name` and `X509_get_subject_name` accept `const X509*`.

Reply:
In OpenSSL 1.0.2 the property accessors just take a non-const `X509` pointer. 
See [x509.h](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/x509/x509.h#L905)

Reply:
Then, I believe, `isSelfSigned` should perform the `const_cast` internally or use the old code for OpenSSL 1.0.x. There is no reason to change the public interface of `isSelfSigned`.

Reply:
Fixed.